src/devices/src/virtio/queue.rs

```diff
@@ -741,6 +765,91 @@ pub(crate) mod tests {
         assert_eq!(q.avail_event(m), 2);
     }
 
+    #[test]
+    #[should_panic(
+        expected = "The number of available virtio descriptors is greater than queue size!"
+    )]
+    fn test_invalid_avail_idx_no_notification() {
+        // This test ensures constructing a descriptor chain succeeds
+        // with valid available ring indexes while it produces an error with invalid
+        // indexes.
+        let m = &create_anon_guest_memory(&[(GuestAddress(0), 0x1000)], false).unwrap();
+
+        // We set up a queue of size 4.
+        let vq = VirtQueue::new(GuestAddress(0), m, 4);
+        let mut q = vq.create_queue();
+
+        q.ready = true;
+
+        for j in 0..4 {
+            vq.dtable[j].set(
+                0x1000 * (j + 1) as u64,
+                0x1000,
+                VIRTQ_DESC_F_NEXT,
+                (j + 1) as u16,
+            );
+        }
```

Comment:
Nit: It is always safe to cast a `u64` into a `usize` but it is not always safe to do the reverse (with stricter linting this would currently raise a warning).
```suggestion
        for j in 0u64..4u64 {
            vq.dtable[usize::from(j)].set(
                0x1000 * (j + 1),
                0x1000,
                VIRTQ_DESC_F_NEXT,
                (j + 1),
            );
        }
```

Reply:
This is a good point @JonathanWoollett-Light! However, this applies to all queue tests in this file. Can you please open an issue for this? Thanks!

Reply:
With a wider context, maybe an issue should relate to adding `#![warn(clippy::as_conversions)]` (https://rust-lang.github.io/rust-clippy/master/)?

Reply:
Or maybe modify add this restriction in the `cargo clippy` command directly if possible.

Reply:
Created basic issue https://github.com/firecracker-microvm/firecracker/issues/3161